plugins/spi/SPIOutput.cpp

```diff
@@ -180,26 +184,86 @@ SPIOutput::SPIOutput(const UID &uid, SPIBackendInterface *backend,
   m_spi_device_name = FilenameFromPathOrPath(m_backend->DevicePath());
 
   PersonalityCollection::PersonalityList personalities;
-  personalities.push_back(Personality(m_pixel_count * WS2801_SLOTS_PER_PIXEL,
-                                      "WS2801 Individual Control"));
-  personalities.push_back(Personality(WS2801_SLOTS_PER_PIXEL,
-                                      "WS2801 Combined Control"));
-  personalities.push_back(Personality(m_pixel_count * LPD8806_SLOTS_PER_PIXEL,
-                                      "LPD8806 Individual Control"));
-  personalities.push_back(Personality(LPD8806_SLOTS_PER_PIXEL,
-                                      "LPD8806 Combined Control"));
-  personalities.push_back(Personality(m_pixel_count * P9813_SLOTS_PER_PIXEL,
-                                      "P9813 Individual Control"));
-  personalities.push_back(Personality(P9813_SLOTS_PER_PIXEL,
-                                      "P9813 Combined Control"));
-  personalities.push_back(Personality(m_pixel_count * APA102_SLOTS_PER_PIXEL,
-                                      "APA102 Individual Control"));
-  personalities.push_back(Personality(APA102_SLOTS_PER_PIXEL,
-                                      "APA102 Combined Control"));
+  // personality description is max 32 characters
+
+  ola::rdm::SlotDataCollection::SlotDataList sd_rgb_combined;
+  sd_rgb_combined.push_back(
+      ola::rdm::SlotData::PrimarySlot(ola::rdm::SD_COLOR_ADD_RED, 0));
+  sd_rgb_combined.push_back(
+      ola::rdm::SlotData::PrimarySlot(ola::rdm::SD_COLOR_ADD_GREEN, 0));
+  sd_rgb_combined.push_back(
+      ola::rdm::SlotData::PrimarySlot(ola::rdm::SD_COLOR_ADD_BLUE, 0));
+  ola::rdm::SlotDataCollection sdc_rgb_combined;
+  sdc_rgb_combined = ola::rdm::SlotDataCollection(sd_rgb_combined);
+
+  ola::rdm::SlotDataCollection::SlotDataList sd_irgb_combined;
+  sd_irgb_combined.push_back(
+      ola::rdm::SlotData::PrimarySlot(ola::rdm::SD_INTENSITY, 0));
```

Comment:
This probably wants to default to 100% aka 255 aka DMX_MAX_SLOT_VALUE, so by default it still just works like RGB mode, but with the option of using the overall intensity.